doc/spec/001-e2e-echo/e2e-echo-full.json5

```diff
@@ -0,0 +1,81 @@
+/*
+ * fh-core offers a special invocation style which can be used
+ * to test the outcome of its processing machinery.
+ *
+ * By invoking the designated HTTP endpoint with a special
+ * query argument like `__fh-echo__=true` or a request header
+ * like `FH-Echo: true`, the machinery will mask calls to
+ * `dispatch_request` and `respond_with` but instead will
+ * record their outcomes and return them as a JSON-serialized
+ * HTTP response.
+ *
+**/
+{
+  "requests": [
+    // Outcome from first invocation of `dispatch_request`.
+    // Derived from `http POST https://httpbin.org/anything foo=bar`.
+    {
```

Comment:
Very much on the spot, thanks. While typing out the spec, that appeared to me as well.

So, we will have to adjust the specification accordingly.

Reply:
So, let's expand this specification beyond its initial scope.

### Auditing
When enabled, recording of all requests and responses will always be enabled as this is something we might want to put into an "audit" database table even in production mode to empower the user to introspect it on all occasions. PutsReq has exactly the same feature, where everything happening will be stored within MongoDB collections.

### Echo mode
This is still what we wrote down in the specification: The final response will be augmented to return everything which has been recorded in order to satisfy an e2e test client.

This mode will get enabled by sending a HTTP header `FH-Echo: true` along with the ingress HTTP request.

### Dry-run mode
That might be an aspect which is orthogonal to the auditing mechanics and the echo mode. Given a scenario where we want to test arbitrary example recipes through an e2e testing machinery, only this mode will mask out calls to `dispatch_request()`, essentially disabling any network communication from the recipe altogether. Obviously, in this mode, there will be no responses to be recorded, so there is nothing we can do about it other than leaving their recording slots empty (`null`).

Also, because no responses are available here, the recipe will not be able to react upon the response content from earlier HTTP requests when running consecutive HTTP conversations subsequently. I know this feels a bit fishy, but nevertheless I wanted to sketch out my thoughts on this.

This mode will get enabled by sending a HTTP header `FH-Dry-Run: true` along with the ingress HTTP request.

Reply:
Sounds like a good starting point! Lets do it exactly as you suggest for now.

Reply:
### Update
We decided to skip the "echo mode" and go for building out the "audit subsystem" instead. This will also benefit users when running the system in production and will hopefully also deliver appropriate features to improve the designated e2e testing environment.

Thus, I will convert this specification pull request back to a draft and will augment it appropriately later, adapting to the updated plan.